warehouse/accounts/views.py

```diff
@@ -169,3 +169,42 @@ def logout(request, redirect_field_name=REDIRECT_FIELD_NAME):
         return HTTPSeeOther(redirect_to, headers=dict(headers))
 
     return {"redirect": {"field": REDIRECT_FIELD_NAME, "data": redirect_to}}
+
+
+@view_config(
+    route_name="accounts.register",
+    renderer="accounts/register.html",
+    decorator=[csrf_protect("accounts.register"), uses_session],
+)
+def register(request, _form_class=forms.RegistrationForm):
+    if request.authenticated_userid is not None:
+        return HTTPSeeOther("/")
+
+    user_service = request.find_service(IUserService, context=None)
+    recaptcha_service = request.find_service(name="recaptcha")
+    recaptcha_service.add_to_csp_policy()
```

Comment:
Actually, maybe a little cleaner would be to implement an `merge` method to the csp service and then just have the recpatcha policy as a constant. With that, from the viwe it would look like:

`request.find_service(name="csp").merge(recaptcha_service.CSP_POLICY)`

Thoughts?